src/util/dom_misc.ts

```diff
@@ -1,301 +1,144 @@
-   * @param {String} attr Style attribute to get for element
-   * @return {String} Style attribute value of the given element.
-   */
-  var getElementStyle;
-  if (fabric.document.defaultView && fabric.document.defaultView.getComputedStyle) {
-    getElementStyle = function(element, attr) {
-      var style = fabric.document.defaultView.getComputedStyle(element, null);
-      return style ? style[attr] : undefined;
-    };
+  if (!doc) {
+    return offset;
   }
-  else {
-    getElementStyle = function(element, attr) {
-      var value = element.style[attr];
-      if (!value && element.currentStyle) {
-        value = element.currentStyle[attr];
-      }
-      return value;
-    };
+  const elemStyle = fabric.document.defaultView.getComputedStyle(element, null)
+  for (const attr in offsetAttributes) {
+    offset[offsetAttributes[attr]] += parseInt(elemStyle[attr], 10) || 0;
   }
 
-  (function () {
-    var style = fabric.document.documentElement.style,
-        selectProp = 'userSelect' in style
-          ? 'userSelect'
-          : 'MozUserSelect' in style
-            ? 'MozUserSelect'
-            : 'WebkitUserSelect' in style
-              ? 'WebkitUserSelect'
-              : 'KhtmlUserSelect' in style
-                ? 'KhtmlUserSelect'
-                : '';
-
-    /**
-     * Makes element unselectable
-     * @memberOf fabric.util
-     * @param {HTMLElement} element Element to make unselectable
-     * @return {HTMLElement} Element that was passed in
-     */
-    function makeElementUnselectable(element) {
-      if (typeof element.onselectstart !== 'undefined') {
-        element.onselectstart = () => false;
-      }
-      if (selectProp) {
-        element.style[selectProp] = 'none';
-      }
-      else if (typeof element.unselectable === 'string') {
-        element.unselectable = 'on';
-      }
-      return element;
-    }
-
-    /**
-     * Makes element selectable
-     * @memberOf fabric.util
-     * @param {HTMLElement} element Element to make selectable
-     * @return {HTMLElement} Element that was passed in
-     */
-    function makeElementSelectable(element) {
-      if (typeof element.onselectstart !== 'undefined') {
-        element.onselectstart = null;
-      }
-      if (selectProp) {
-        element.style[selectProp] = '';
-      }
-      else if (typeof element.unselectable === 'string') {
-        element.unselectable = '';
-      }
-      return element;
-    }
+  const docElem = doc.documentElement;
+  if ( typeof element.getBoundingClientRect !== 'undefined' ) {
+    box = element.getBoundingClientRect();
+  }
 
-    fabric.util.makeElementUnselectable = makeElementUnselectable;
-    fabric.util.makeElementSelectable = makeElementSelectable;
-  })(typeof exports !== 'undefined' ? exports : window);
+  const scrollLeftTop = getScrollLeftTop(element);
 
-  function getNodeCanvas(element) {
-    var impl = fabric.jsdomImplForWrapper(element);
-    return impl._canvas || impl._image;
+  return {
```

Comment:
Point?

Reply:
better off for rtl support as well?

Reply:
i think those are just for calculating the canvas position for some sort of padding/border issue.
I m not sure we ever need a different measurement.

I also think modern browsers solves this for us giving us an extra pair of mouse event coordinates relative to the canvas element that would be exactly what we need

Reply:
yes i think that i should look into moving mouse events with offsetX and offsetY and trash all of this.

Reply:
i ll open a task